pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/filter/DsmAuthFilter.java

```diff
@@ -93,14 +94,21 @@ boolean isTokenValueValid(String tokenValue) {
      * Get the clientId reported by the token.
      *
      * @param jwt the decoded token object
-     * @return the clientId extracted from token "sub" claim
+     * @return the clientId extracted from token client claim
      */
     private Optional<String> extractClientIdFromToken(DecodedJWT jwt) {
-        //sub claim contains the clientId with a suffix. Extract the client id from it
-        return Optional.ofNullable(jwt.getClaim("sub"))
-                .map(claim -> claim.asString().substring(0, claim.asString().indexOf("@clients")));
+        Optional<String> clientId;
+        clientId = Optional.ofNullable(jwt.getClaim(Auth0Constants.DDP_CLIENT_CLAIM).asString());
+        if (!clientId.isPresent()) {
+            //check if sub claim contains the clientId with a suffix.
+            //backward compatibility until we see new DSM Tokens
+            clientId = Optional.ofNullable(jwt.getClaim("sub"))
```

Comment:
DSM backend currently sends claims in "sub" . Once DSM changes are deployed and DSM starts sending new DSM Tenant Tokens, this additional fall back can be cleaned up.
Added now for backward compatibility